[hipFile] first pass at documentation (#6747)
ROCm/rocm-systems: projects/hipfile/docs/install/build-from-source.rst

*************************
Build hipFile from source
*************************

Prerequisites
*************

Before building hipFile, make sure you have the following software installed:

- ``ROCm`` (provides ``hsa-runtime64`` and HIP). See :doc:`Install AMD ROCm <rocm:install/rocm>` for current requirements.
- ``CMake`` >= 3.21
- C++17 compiler (Clang or GCC)
- ``libmount`` (from ``util-linux``)

Install the ROCm Core SDK
*************************

hipFile is included with the ROCm Core SDK. For the most complete installation, use the ``amdrocm-core-sdk`` meta package. Refer to :doc:`Install AMD ROCm <rocm:install/rocm>` for detailed installation instructions.

Install on Linux
****************

To install only the hipFile library group as a subset of the ROCm Core SDK, use the appropriate ``amdrocm-*`` meta package. The package name follows this format:

.. code-block:: text

   amdrocm-<group><-dev/-devel><rocm_version><-llvm_target>

Where:

- ``-dev`` is used for Debian-based distributions
- ``-devel`` is used for RPM-based distributions
- ``<rocm_version>`` is an optional version pin
- ``<-llvm_target>`` is an optional single GPU architecture specifier

.. note::

>    Make sure the exact package name for hipFile with the latest ROCm release documentation, as the group name may vary between releases.

Reviewer: This sentence is missing its verb. (Also "may" → "might" per our style.)

```suggestion
   Verify the exact package name for hipFile against the latest ROCm release documentation, as the group name might vary between releases.
```

.. tab-set::

   .. tab-item:: Debian-based

      .. code:: shell

         sudo apt-get update
         sudo apt-get install amdrocm-<group>-dev

   .. tab-item:: RHEL-based

      .. code:: shell

         sudo dnf install amdrocm-<group>-devel

   .. tab-item:: SLES

      .. code:: shell

         sudo zypper install amdrocm-<group>-devel

Building from source
********************

Source download
---------------

Clone the repository using a sparse checkout from the ROCm monorepo:

.. code:: shell

   git clone --filter=blob:none --sparse https://github.com/ROCm/rocm-systems.git
   cd rocm-systems
   git sparse-checkout set projects/hipfile

Library dependencies
--------------------

hipFile requires the following libraries during configuration and linking:

- ``hsa-runtime64``: HSA runtime (found via CMake config)
- ``hip``: HIP runtime (found via CMake config)
- ``libmount``: mount information parsing (from ``util-linux``; located via ``find_library``)

Build commands
--------------

Create a build directory, configure with CMake, build, and install:

> On the NVIDIA platform, these additional dependencies are required:

Reviewer: This whole Build-commands body is a generation artifact — the NVIDIA dependency note repeats **14×**, the NVIDIA build block **3×**, and the AMD build block ended up *after* all of them. It should collapse back to the intended AMD-then-NVIDIA structure below (covers roughly lines 93–203). Too large for a clean inline suggestion, so easiest to apply manually.

```rst
.. code:: shell

   cd rocm-systems/projects/hipfile
   cmake -B build \
       -DCMAKE_INSTALL_PREFIX=/opt/rocm \
       -DCMAKE_HIP_PLATFORM=amd
   cmake --build build
   sudo cmake --install build

On the NVIDIA platform, these additional dependencies are required:

- CUDAToolkit: CUDA runtime and headers
- cuFile: NVIDIA GPUDirect Storage library

For an NVIDIA platform build, set ``CMAKE_HIP_PLATFORM`` to ``nvidia`` and make sure
``CUDAToolkit`` and ``cuFile`` are discoverable:

.. code:: shell

   cmake -B build \
       -DCMAKE_HIP_PLATFORM=nvidia \
       -DCMAKE_INSTALL_PREFIX=/usr/local
   cmake --build build
   sudo cmake --install build
```

Author: good catch! I thought I had gotten most of the nvidia references.
